python/lsst/analysis/drp/dataSelectors.py

```diff
@@ -151,3 +164,41 @@ def __call__(self, df, **kwargs):
         sourceType = np.zeros(len(df))
         sourceType[gals] = 2
         return sourceType
+
+
+class UnknownIdentifier(DataFrameAction):
+    """Identifies un classified objects from the dataFrame and marks them as a
+       9 in the added sourceType column"""
+
+    band = Field(doc="The band the object is to be classified as an unkown in.",
+                 default="i",
+                 dtype=str)
+
+    @property
+    def columns(self):
+        return [self.band + "Extendedness"]
+
+    def __call__(self, df, **kwargs):
+        """Identifies sources classed as unknowns"""
+        unknowns = (df[self.band + "Extendedness"] == 9.0)
+        sourceType = np.zeros(len(df))
+        sourceType[unknowns] = 9
+        return sourceType
+
+
+class CoaddPlotFlagSelector(FlagSelector):
+    """The flags to use for selecting sources for coadd QA"""
+
+    bands = ListField(doc="The bands to apply the flags in",
+                      dtype=str,
+                      default=["g", "r", "i", "z", "y"])
+
+    def setDefaults(self):
+        super().setDefaults()
+        # This is missing one of the PA flags because they are not
+        # currently in the objectTablesbase_SdssCentroid_flag, is it
+        # redundant with xy_flag?
+        flagCols = ["PsfFlux_flag", "PixelFlags_saturatedCenter", "Extendedness_flag"]
+
+        filterColumns = ["xy_flag"] + [band + flag for flag in flagCols for band in self.bands]
```

Comment:
My understanding is that `setDefaults()` gets called before the user's configuration is applied, so in this function self.bands is always grizy and the config in the pipeline yaml never gets propagated to `self.selectWhenFalse`. (When I ran `coaddQAPlots.yaml` on a collection with only r and i-band data, all the plots were empty because the selector was seeing flags set on z and y band columns.)

Reply:
Thank you for catching this! Some faffing later this is fixed.

Reply:
Excellent, I can confirm that the selector is now picking up the right bands. Plots of two-band data are still coming up empty for me though, and I think it's because the default `CoaddPlotFlagSelector` is still being applied (which requires grizy), in addition to the other selectors specified in the config yaml. I guess the config treats the additional selectors as an append rather than a replacement of the default?

(I didn't mean to pick such an edge case to start with, this is just the  `pipelines_check` data.)

Reply:
Edge cases are good.

Reply:
Hey look, if I wait long enough to do a review, Colin finds broken things for me and Sophie fixes them! 😂

Reply:
Sweet, I have plots now! This is very exciting, and I will leave Meredith with the actual review (I'm not going to take all the work 🙂). There does seem to be a few places where the code throws un-graceful exceptions when no objects exist or when various selection arrays are empty, edge cases like that. Trying out a few less-than-fully-complete datasets like `pipelines_check` (or temporarily adding overly-stringent selectors) would show where additional guards would help.

Reply:
I will add more testing to the list of things to do once I have a few more plot types going.